libgnucash/app-utils/gnc-ui-util.c

```diff
@@ -1863,6 +1865,53 @@ xaccPrintAmount (gnc_numeric val, GNCPrintAmountInfo info)
     return buf;
 }
 
+const char *
+gnc_print_amount_with_bidi_ltr_isolate (gnc_numeric val, GNCPrintAmountInfo info)
+{
+    /* hack alert -- this is not thread safe ... */
+    static char buf[BUFLEN];
+    static const char ltr_isolate[] = { 0xe2, 0x81, 0xa6 };
+    static const char ltr_pop_isolate[] = { 0xe2, 0x81, 0xa9 };
+    size_t offset = info.use_symbol ? 3 : 0;
+
+    memset (buf, 0, BUFLEN);
+    if (!xaccSPrintAmount (buf + offset, val, info))
+    {
+        buf[0] = '\0';
+        return buf;
+    };
+
+    if (!info.use_symbol)
+        return buf;
+
+    memcpy (buf, ltr_isolate, 3);
+
+    if (buf[BUFLEN - 1] != '\0')
+    {
+        buf[BUFLEN - 1] = '\0';
+        memcpy (buf + BUFLEN - 4, ltr_pop_isolate, 3);
+    }
+    else
+    {
+        size_t length = strlen (buf);
+        memcpy (buf + length, ltr_pop_isolate, 3);
```

Comment:
Oops, this could overrun the buffer and fail to be null-terminated if `strlen(buf) > BUFLEN - 4`.

To fix the first test could be `if ((uint32_t)buf[BUFLEN - 4])`. You'll need to add `#include <stdint.h>` up top. The reason for `uint32_t` is that Debian builds for a bunch of weird architectures like IBM360 and I'm not sure that `int` is 32 bits on all of them; and even if it is this year that can change as `sizeof(int)` is implementation-defined.

Reply:
I have changed it differently, so the if test is `if (buf[BUFLEN - 4] == '\0')` and now there should be room for the pop_isolate,
If the test fails, I set the last element of the buffer to `\0` and then `memcpy (buf + BUFLEN - 4, ltr_pop_isolate, 3)`. This obviously truncates the output but should never happen with the buffer size of 1024!
I did wonder about adding a PWARN for this but not sure it is worth it.

Reply:
I think skip the test and always assign. 

Considering how unlikely it is that we'd see a 1017-byte representation of a number, hitting this branch means that we've somehow gotten a non-numeric string out of xaccSPrintAmount. That probably does justify a PWARN.

Reply:
Suggested changes made.